File: Per_Diem_app - Beta/app/src/main/java/com/example/jay/theperdiemapp/models/Report.java
package com.example.jay.theperdiemapp.models;

import java.io.Serializable;
import java.util.ArrayList;

public class Report implements Serializable {

    private String name;
    private String startDate;
    private String endDate;
    private String memo;
    private String reportID;
    private String status;
    private Double total;


    public Report (){}

    public Report (String Name, String StartDate, String EndDate){
        Long tsLong = System.currentTimeMillis()/1000;
        String ts = tsLong.toString();

        this.name = Name;
        this.startDate = StartDate;
        this.endDate = EndDate;
        this.reportID = Name+ts;
        this.memo = "";
        this.total = 0.00;
        this.status = "Created";

    }

    public Report (String Name, String StartDate, String EndDate, String Memo){
        Long tsLong = System.currentTimeMillis()/1000;
        String ts = tsLong.toString();

        this.name = Name;
        this.startDate = StartDate;
        this.endDate = EndDate;
        this.memo = Memo;
        this.reportID = Name+ts;
        this.total = 0.00;
        this.status = "Created";
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public String getMemo() {
        return memo;
    }

    public void setMemo(String memo) {
        this.memo = memo;
    }

    public String getReportID() {
        return reportID;
    }

    public void setReportID(String reportID) {
        this.reportID = reportID;
    }

    public Double getTotal() {
        return total;
    }

    public void setTotal(Double total) {
        this.total = total;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: IOS Version/Per Diem App/Per Diem App/Models/Report.swift
//
//  Report.swift
//  Per Diem App
//
//  Created by Jason Claxton on 7/6/18.
//  Copyright © 2018 Jason Claxton. All rights reserved.
//

import Foundation
import UIKit

class Report {


    var name:String
    var startDate:String
    var endDate:String
    var memo:String
    var reportID:String
    var status:String
    var total:Double
    
    init(Name:String?,StartDate:String?,EndDate:String?,Memo:String?,ReportID:String?,
         Status:String?,Total:Double?) {
        self.name = Name ?? ""
        self.startDate = StartDate ?? ""
        self.endDate = EndDate ?? ""
        self.memo = Memo ?? ""
        self.reportID = ReportID ?? ""
        self.status = Status ?? ""
        self.total = Total!
        
    }

}
